proposals/partial-type-inference.md

```diff
@@ -0,0 +1,824 @@
+  
+  * We use the previous example.
+    After the first phase. `T1` has bounds `IList<_>` and `List<int>`. `_` has bound `int`.
+    In this situation, we can't start to fix `T1` because `_` is not fixed yet.
+    `T1` is type-dependent on `_`.
+    So, we will first fix `_`, which becomes `int`.
+    Then, `T1` is not type-dependent anymore, because all bounds don't contain any unfixed type variables.
+    `IList<_>` is now `IList<int>` after the `_` fixation.
+    We can fix `T1` now.
+
+* Explanation of shape-dependence
+  * A similar thing is for shape-dependence.
+    Although it cares about bounds received from the type argument list.
+    We want a shape bound to be exact (not containing any unfixed type variables) because it is later important for the fixation.
+    An intention is to keep the exact form of the given hint(`IList<_>`).
+
+  * Example: Given `IList<_>` as a type argument, when we treat nullability, we want the hinted type parameter to be non-nullable(not `IList<_>?`).
+    It can happen, other bounds would infer the nullable version, and although `IList<_>` can be converted to `IList<_>?`, it is not the user's intention.
+
+* Explanation of inference restriction during constructor type inference
+  * Because performing type inference can even take exponential time when a type system contains overloading, the restriction was made above to avoid it. 
+    It regards binding arguments before the overload resolution when we bind all *object_creation_expressions* without target info and then in case of overload resolution success and some of these arguments failed in the binding, we try to bind it again with already known target type information.
+
+### Compile-time checking of dynamic member invocation
+
+We change the [compile-time checking](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#1265-compile-time-checking-of-dynamic-member-invocation) in order to be useful during partial type inferece.
+
+- First, if `F` is a generic method and type arguments were provided, then those ***, that aren't *inferred_type_argument*** are substituted for the type parameters in the parameter list. However, if type arguments were not provided, no such substitution happens.
+- Then, any parameter whose type is open (i.e., contains a type parameter; see §8.4.3) is elided, along with its corresponding parameter(s).
+- **If the method group contains inferred type arguments, a warning should appear since partial type inference is not supported by runtime.** 
+
+We add the following
+- If an object creation expression is inferred and the argument list contains a dynamic value, an error should appear since constructor type inference is not supported by runtime.  
+
+### Nullability
+
+We add the following section about nullability. 
+
+* We can use an examination mark `?` to say that the inferred type argument should be a nullable type (e.g. `F<_?>(...)`).
```

Comment:
```suggestion
* We can use a question mark `?` to say that the inferred type argument should be a nullable type (e.g. `F<_?>(...)`).
```
Also, if this is intended to update a particular section of the language standard, let's link to that section here.

Reply:
Solved by commit [98d835e](https://github.com/dotnet/csharplang/pull/7582/commits/98d835e3dcbd67efbb7ae1c87cf54ac19578d68f)